### DISCUSSION TITLE
Como ejecutar los tests.

### DISCUSSION THREAD

**Discussion by MiguelUlloaWood** [IIC2133-PUC/2024-1#5] [Tarea 0] [answered]
Hola, quería saber si hay alguna guía sobre como ejecutar los tests.

**Comment by PabloNavarroY** [comment, marked as answer]
Como aparece en el enunciado, tu programa debe ejecutarse con el comando
```sh
./ship path/to/input path/to/output
```
y esto producirá o modificará el archivo `path/to/output`, el cual puedes comparar con el archivo `correct-output.txt` que viene con cada test.

Ahora, si estás en Linux, MacOs o tienes WSL instalado puedes ocupar el programa [`diff`](https://www.man7.org/linux/man-pages/man1/diff.1.html) para comparar dos archivos. Solo tienes que poner en la terminal
```sh
diff path/to/file1 path/to/file2
```
y aparecerán todas las diferencias entre los dos archivos. Si los dos archivos son iguales esto no producirá ningún output.

**Notas aparte**
- Si estás ocupando Windows con WSL, asegúrate de ocupar `diff` en la terminal de WSL (bash), no en powershell o cmd.
- Puedes [comparar archivos en VSCode](https://vscode.one/diff-vscode/).
- Todo este proceso de comparar archivos se puede automatizar con shell scripting, pero no tenemos una guía sobre eso y te recomiendo terminar la tarea antes de investigar sobre esto.
- Dependiendo del contenido de los enunciados de las proximas tareas, puede que los tests de estas no tengan un "output correcto", si no que una serie de outputs validos. Esto no es importante ahora, pero si lo puede ser en un futuro.
